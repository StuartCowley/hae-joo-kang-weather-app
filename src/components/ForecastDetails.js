import React from "react";
import PropTypes from "prop-types";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();
  return (
    <div className="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__temperature">
        <div>Max Temperature: {temperature.max}&deg;C</div>
        <div>Min Temperature: {temperature.min}&deg;C</div>
      </div>
      <div className="forecast-details__humidity">humidity: {humidity}&#37;</div>
      <div className="forecast-details__wind">wind: {wind.speed}mph {wind.direction}</div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
