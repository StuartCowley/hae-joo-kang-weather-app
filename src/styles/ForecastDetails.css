.forecast-details {
  margin-top: 30px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: 220px;
  height: 180px;
  background-color: rgba(255, 255, 255, 0.3);
  box-shadow: rgba(0, 0, 0, 0.15) 0px 10px 10px 0px;
  padding: 10px;
}