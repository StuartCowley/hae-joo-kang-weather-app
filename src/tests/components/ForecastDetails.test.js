import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    forecast: {
      date: 1525046400000,
      temperature: {
        min: 4,
        max: 11,
      },
      humidity: 30,
      wind: {
        speed: 13,
        direction: "s",
      },
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails forecast={validProps.forecast} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
