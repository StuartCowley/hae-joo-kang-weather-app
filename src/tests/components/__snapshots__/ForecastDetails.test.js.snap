// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__temperature"
    >
      <div>
        Max Temperature: 11°C
      </div>
      <div>
        Min Temperature: 4°C
      </div>
    </div>
    <div
      class="forecast-details__humidity"
    >
      humidity: 30%
    </div>
    <div
      class="forecast-details__wind"
    >
      <span
        class="forecast-details__wind--speed"
      >
        wind: 13mph 
      </span>
      <span
        class="forecast-details__wind--direction wi wi-wind wi-from-s"
      />
    </div>
  </div>
</DocumentFragment>
`;
